src/python/WMCore/ReqMgr/CherryPyThreads/HeartbeatMonitor.py

```diff
@@ -1,5 +1,187 @@
+               "requestsByStatusAndPrio": {},
+               "requestsByStatusAndNumEvts": {}}
+
+    for st in ACTIVE_STATUS:
+        results["requestsByStatus"].setdefault(st, 0)
+        results["requestsByStatusAndNumEvts"].setdefault(st, 0)
+        results["requestsByStatusAndCampaign"].setdefault(st, {})
+        results["requestsByStatusAndPrio"].setdefault(st, {})
+
+    return results
+
 
 class HeartbeatMonitor(HeartbeatMonitorBase):
-    pass
+
+    def __init__(self, rest, config):
+        super(HeartbeatMonitor, self).__init__(rest, config)
+        self.userAMQ = getattr(config, "user_amq", None)
+        self.passAMQ = getattr(config, "pass_amq", None)
+
+        self.producer = "reqmgr2"
+        self.docTypeAMQ = "cms_%s_info" % self.producer
+
+    def addAdditionalMonitorReport(self, config):
+        """
+        _addAdditionalMonitorReport_
+
+        Collect general request information and post it to both central
+        couchdb and MonIT. Items to fetch:
+         * number of requests in each status (excluding archived ones)
+         * for requests in assignment-approved:
+           * their priority
+           * their RequestNumEvents (if there is no input dataset)
+         * for assignment-approved, assigned, acquired, running-* and completed:
+           * number of requests in each campaign
+        """
+        self.logger.info("Collecting ReqMgr2 statistics...")
+        wmstatsSvc = WMStatsServer(config.wmstatsSvc_url, logger=self.logger)
+
+        results = initMetrics()
+
+        inputConditon = {}
+        outputMask = ["RequestStatus", "RequestType", "RequestPriority",
+                      "Campaign", "RequestNumEvents"]
+        startT = int(time.time())
+        for reqInfo in wmstatsSvc.getFilteredActiveData(inputConditon, outputMask):
+            status = reqInfo['RequestStatus']
+            results['requestsByStatus'][status] += 1
+
+            for campaign in _getCampaign(reqInfo["Campaign"]):
+                results["requestsByStatusAndCampaign"][status].setdefault(campaign, 0)
+                results['requestsByStatusAndCampaign'][status][campaign] += 1
+
+            requestPrio = reqInfo['RequestPriority']
+            results["requestsByStatusAndPrio"][status].setdefault(requestPrio, 0)
+            results['requestsByStatusAndPrio'][status][requestPrio] += 1
+
+            results['requestsByStatusAndNumEvts'][status] += _getRequestNumEvents(reqInfo['RequestNumEvents'])
+
+        endT = int(time.time())
+        results["total_query_time"] = endT - startT
+
+        if self.postToAMQ:
+            allDocs = self.buildMonITDocs(results)
+            self.uploadToAMQ(allDocs)
+
+        return results
+
+    def buildMonITDocs(self, stats):
+        """
+        Given the statistics that are uploaded to wmstats, create different
+        documents to post to MonIT AMQ (aggregation-friendly docs).
+        """
+        commonInfo = {"version": '0.3',
```

Comment:
where is the version comes from?

Reply:
Good question :) I took it from Benjamin's script, however I think we don't really need it. StompAMQ has already a version in the headers. Anyways, we can keep it for now, getting info from CERN IT is a great challenge...